tests/testsuite/freshness_checksum.rs

```diff
@@ -0,0 +1,2880 @@
+//! Tests for fingerprinting (rebuild detection).
+
+#![allow(deprecated)]
+
+use std::fs::{self, OpenOptions};
+use std::io::prelude::*;
+use std::net::TcpListener;
+use std::process::Stdio;
+use std::thread;
+
+use super::death;
+use cargo_test_support::paths::CargoPathExt;
+use cargo_test_support::prelude::*;
+use cargo_test_support::registry::Package;
+use cargo_test_support::{basic_lib_manifest, basic_manifest, project, rustc_host, rustc_host_env};
+
+#[cargo_test]
+fn checksum_actually_uses_checksum() {
+    let p = project()
+        .file("src/main.rs", "mod a; fn main() {}")
+        .file("src/a.rs", "")
+        .build();
+
+    p.cargo("build")
+        .masquerade_as_nightly_cargo(&["checksum-freshness"])
+        .args(&["-Z", "unstable-options", "-Z", "checksum-freshness"])
```

Comment:
nit: could you do
```rust
p.cargo("build -Zchecksum-freshness")
```

Identifying quickly in the test case what command is being run makes it a lot easier to read the test.  This buries it compared to most tests.

Reply:
Resolved in 3cc6814ee8ede4ab4f3defb581f37c0ae1a1623a